linkerd/protocol/http/src/e2e/scala/io/buoyant/linkerd/protocol/HttpEndToEndTest.scala

```diff
@@ -693,4 +697,152 @@ class HttpEndToEndTest
+
+      // An incoming request through the Http.Router will establish an active connection; We expect to see it here
+      assert(activeConnectionsCount() == 1.0)
+
+      eventually(timeout(Span(5, Seconds)), interval(Span(250, Millis))) {
+        val cnt = activeConnectionsCount()
+        assert(cnt == 0.0)
+      }
+
+      val clientSessionParams = router.params[PerClientParams].paramsFor(Path.read("/svc/century"))[ExpiringService.Param]
+      assert(clientSessionParams.idleTime == 1500.milliseconds)
+      assert(clientSessionParams.lifeTime == Duration.Top)
+
+      ()
+    }
+  }
+
+  test("clientSession idleTimeMs should not close client connections when isn't specified") {
+    val config =
+      s"""|routers:
+          |- protocol: http
+          |  experimental: true
+          |  dtab: |
+          |    /p/fox => /$$/inet/127.1/{fox.port} ;
+          |    /svc/century => /p/fox ;
+          |  servers:
+          |  - port: 0
+          |  client:
+          |    forwardClientCert: false
+          |""".stripMargin
+
+    idleTimeMsBaseTest(config){ (router:Router.Initialized, stats:InMemoryStatsReceiver, foxPort:Int) =>
+
+      // Assert
+      def activeConnectionsCount = stats.gauges(Seq("rt", "http", "client", s"$$/inet/127.1/${foxPort}", "connections"))
+
+      // An incoming request through the Http.Router will establish an active connection; We expect to see it here
+      assert(activeConnectionsCount() == 1.0)
+
+      val clientSessionParams = router.params[PerClientParams].paramsFor(Path.read("/svc/century"))[ExpiringService.Param]
+      assert(clientSessionParams.idleTime == Duration.Top)
+      assert(clientSessionParams.lifeTime == Duration.Top)
+
+      ()
+    }
+  }
+
+  def idleTimeMsBaseTest(config:String)(assertionsF: (Router.Initialized, InMemoryStatsReceiver, Int) => Unit): Unit = {
+    // Arrange
+    val stats = new InMemoryStatsReceiver
+    val fox = Downstream.const("fox", "what does the fox say?")
+
+    val configWithPort = config.replace("{fox.port}", fox.port.toString)
+
+    val linker = Linker.Initializers(Seq(HttpInitializer)).load(configWithPort)
+      .configured(param.Stats(stats))
+
+    val router = linker.routers.head.initialize()
+    val server = router.servers.head.serve()
+
+    val client = upstream(server)
+
+    def get(host: String, path: String = "/")(f: Response => Unit): Unit = {
+      val req = Request(path)
+      req.host = host
+      val rsp = await(client(req))
+      f(rsp)
+    }
+
+    // Act
+    try {
+      get("century") { rsp =>
+        assert(rsp.status == Status.Ok)
+        assert(rsp.contentString == "what does the fox say?")
+        ()
+      }
+
+      // Assert
+      assertionsF(router, stats, fox.port)
+
+    } finally {
+      await(client.close())
+      await(fox.server.close())
+      await(server.close())
+      await(router.close())
+    }
+
+
+  }
```

Comment:
I may be wrong, but maybe we could add a test that exercises both `hostConnectionPool` and `clientSession` configs. I am still unclear of how the two different kinds of configurations would work together if that ever is the case.

Reply:
@dadjeibaah , I can look into adding more tests for the scenarios when both configurations are used. But I will be able to do that only starting Tuesday (24th) next week.